docs/snapshot.md

```diff
@@ -0,0 +1,154 @@
+
+```yaml
+apiVersion: topolvm.cybozu.com/v1
+kind: LogicalVolume
+metadata:
+  name: snapcontent-b083470e-8293-47cc-810d-9561bd1754e6 # volumesnapshotcontent name
+  annotations:
+    topolvm.cybozu.com/source: df4b5a3e-8285-4554-b964-8aec23339be1 # source-pv.csi.volumeHandle
+spec:
+  name: snapcontent-b083470e-8293-47cc-810d-9561bd1754e6
+  nodeName: 192.168.26.40
+  size: 1Gi
+```
+
+### logicalvolume service
+
+add api
+
+```
+func (s *LogicalVolumeService) CreateSnapshot(ctx context.Context, snapName string, sourceVolume *topolvmv1.LogicalVolume) (string, error)
+
+```
+
+### logicalvolume-controller
+
+check logicalvolume(CR) has annotation(topolvm.cybozu.com/source) or not. if yes, call lvmd rpc api (CreateSnapshot)
+
+### lvmd
+
+add CreateSnapshot RPC api
+
+```
+
+type CreateSnapshotRequest struct {
+    state         protoimpl.MessageState
+    sizeCache     protoimpl.SizeCache
+    unknownFields protoimpl.UnknownFields
+     
+    Name        string   `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`                    // The logical volume name.
+    SizeGb      uint64   `protobuf:"varint,2,opt,name=size_gb,json=sizeGb,proto3" json:"size_gb,omitempty"` // Volume size in GiB.
+    Tags        []string `protobuf:"bytes,3,rep,name=tags,proto3" json:"tags,omitempty"`                    // Tags to add to the volume during creation
+    DeviceClass string   `protobuf:"bytes,4,opt,name=device_class,json=deviceClass,proto3" json:"device_class,omitempty"`
+    Source      string   `protobuf:"bytes,5,opt,name=source,json=source,proto3" json:"source,omitempty"` // source lv name
+}
+```
+
+lvmd will call lvcreate to create snapshot
+
+```shell
+ lvcreate --snapshot --permission r --size {SizeGb} --name {name} {source}
+```
+
+
+
+Delete Snapshot
+----------
+![component diagram](img/snapshot-delete.svg)
+
+volumesnapshot(CR) delete operation triggers logicalvolume deletion.  
+
+
+Restore Snapshot
+-----------
+![component diagram](img/snapshot-restore.svg)
+
+
+the size and the volume group of restore lv is the same with origin lv.
+
+pvc restore from snapshot, the StorageClass's volumeBindingMode must be WaitForFirstConsumer. make sure restored logicalvolume and snapshot logicalvolume in the same node.
+so the pvc controller will check pvc has datasource or not. if source is snapshot, pvc controller will get the node of snapshot and patch the annotation of selected node to pvc.And then wait for the restore done.  
+
+```yaml
+apiVersion: topolvm.cybozu.com/v1
+kind: LogicalVolume
+metadata:
+  name: pvc-42041aee-79a2-4184-91aa-5e1df6068b9f # restore pv name
+  annotations:
+    topolvm.cybozu.com/snapshot: df4b5a3e-8285-4554-b964-8aec23339be1 # snapshotVolID
+    topolvm.cybozu.com/volumemode: Filesystem   # Filesystem or Block
+    topolvm.cybozu.com/fstype: xfs # xfs or ext4
+spec:
+  name: pvc-42041aee-79a2-4184-91aa-5e1df6068b9f
+  nodeName: 192.168.26.40
+  size: 1Gi
+```
+
```

Comment:
### Validate pods having multiple PVCs that are restored from snapshots.

Users can apply a pod having some PVCs that are restored from snapshots. If users specify PVCs with snapshots placed on a different node as the data sources, the restore will fail and the pod will not start. I think it is better to validate it before applying manifests using a validation webhook. This feature is not a requirement, but I would like to implement this feature.